parse/process.go

```diff
@@ -43,18 +43,16 @@ func Process(r io.Reader, optionsFunc ...OptionsFunc) (*GoTestSummary, error) {
 
 			badLines++
 			if started || badLines > 50 {
-				switch err.(type) {
-				case *json.SyntaxError:
-					err = fmt.Errorf("line %d json error: %s: %w", badLines, err.Error(), ErrNotParseable)
+				var syntaxError *json.SyntaxError
+				if errors.As(err, &syntaxError) {
```

Comment:
I personally prefer in-lining the error so it doesn't leak out of the block, e.g.,

```go
if syntaxError := (*json.SyntaxError)(nil); errors.As(err, &syntaxError) {
  // can only use syntaxError in this block
}
```

But this also works.

Reply:
Indeed, I often forgot it can be done

I would have used this if I thought about it, because I find the syntax more readable, then nil pointer

```go
if syntaxError := new(json.SyntaxError); errors.As(err, &syntaxError) {

}
```

Reply:
@mfridman you never came back to me about this suggestion.

Not that's blocking for the PR, it's merged after all. But I'm curious to know what you think about it

Reply:
Either works fine, as does `if syntaxError := &json.SyntaxError{} ...`

The nil pointer probably doesn't allocate, so it's _marginally_ better? but reads worse. The thing I can more about usually is readability and correctness, it just so happened I could used to writing out the nil pointer way.